src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -216,6 +216,141 @@ OMapInnerNode::rm_key(omap_context_t oc, const std::string &key)
   });
 }
 
+OMapInnerNode::rm_key_range_ret
+OMapInnerNode::rm_key_range(omap_context_t oc, const std::string &first, const std::string &last)
+{
+  LOG_PREFIX(OMapInnerNode::rm_key_range);
+  DEBUGT("remove keys: {}--{}, *this: {}", oc.t, first, last, *this);
+  auto child_fiter = first == ""? iter_cbegin() : get_containing_child(first);
+  auto child_liter = last == ""? (iter_cend()-1) : get_containing_child(last);
+  assert(child_fiter != iter_cend());
+
+  return seastar::do_with(
+    child_fiter,
+    child_liter,
+    iter_t(child_fiter),
+    mutation_result_t(mutation_status_t::SUCCESS,
+      std::nullopt, std::nullopt),
+    std::map<uint16_t, OMapNodeRef>{},
+    [this, oc, &first, &last] (
+      auto &fiter,
+      auto &liter,
+      auto &iter,
+      auto &ret,
+      auto &m_entries)
+    {
+      return trans_intr::repeat(
+        [&, oc, this]() -> rm_key_range_iertr::future<seastar::stop_iteration>
+      {
+        if (iter > liter) {
+          return rm_key_range_iertr::make_ready_future<seastar::stop_iteration>(
+            seastar::stop_iteration::yes);
+        }
+        return get_child_node(oc, iter)
+          .si_then([&, oc, this](auto &&extent) {
+          return seastar::do_with(
+            iter == fiter ? first : std::string{},
+            iter == liter ? last : std::string{},
+            [&, extent = std::move(extent)] (auto &first, auto &last) {
```

Comment:
To make it easier to distinguish from do_with parameters and the nested logic, I think this might be better:
```c++
return seastar::do_with(
  ...
  []()
{  // helps to visually split the long parameters above and the long nested logic below
  <.. 2-space nested code ..>
}); // nesting is finished and the indention recovers at the same time.
```